src/details/ArborX_MinimumSpanningTree.hpp

```diff
@@ -612,13 +613,62 @@ void computeParents(ExecutionSpace const &space, Edges const &edges,
           key = INT_MAX;
 
         // Comparison of weights as ints is the same as their comparison as
-        // floats as long as they are positive and are not NaNs or inf
+        // floats as long as they are positive and are not NaNs or inf.
         static_assert(sizeof(int) == sizeof(float));
-        keys(e) = (key << shift) + KokkosExt::bit_cast<int>(edge.weight);
+        keys(e) = (key << shift) +
+                  (edge.weight ? KokkosExt::bit_cast<int>(edge.weight) : 0);
```

Comment:
Wait what?
isn't `bit_cast<int>(0.f)` equal to `0`?

Reply:
https://godbolt.org/z/93ssn1Tf3

Reply:
I thought maybe to be a bit more careful, as the bit cast of a negative 0 is not 0. But that shouldn't happen in practice, unless someone does some weird distance calculations. 

Would you prefer to change it back?

Reply:
I don't fully get it.  Let's talk.